CMakeLists.txt

```diff
@@ -60,9 +60,14 @@ project (Jerry CXX C ASM)
 # Defining options
  option(ENABLE_VALGRIND       "Enable valgrind helpers in memory allocators" OFF)
  option(ENABLE_VALGRIND_FREYA "Enable valgrind-freya helpers in memory allocators" OFF)
- option(ENABLE_LTO            "Enable LTO build" ON)
  option(ENABLE_LOG            "Enable LOG build" OFF)
- option(ENABLE_ALL_IN_ONE     "Enable ALL_IN_ONE build" OFF)
+ if("${PLATFORM}" STREQUAL "DARWIN")
+  option(ENABLE_LTO            "Enable LTO build" OFF)
+  option(ENABLE_ALL_IN_ONE     "Enable ALL_IN_ONE build" ON)
```

Comment:
Would it be possbile to have this darwin case moved into the other elseif (... Darwin) which is down a few lines from here and have here only the default option? If that's not possible then the PR is ok for me. However we should investigate that is this only Mac related or clang related issue (but that's another topic for now imho)